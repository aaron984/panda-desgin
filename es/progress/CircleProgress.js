import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import React from 'react';

var Circle =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Circle, _React$Component);

  function Circle() {
    _classCallCheck(this, Circle);

    return _possibleConstructorReturn(this, _getPrototypeOf(Circle).apply(this, arguments));
  }

  _createClass(Circle, [{
    key: "getPathStyles",
    value: function getPathStyles() {
      var _this$props = this.props,
          percent = _this$props.percent,
          strokeWidth = _this$props.strokeWidth,
          gapDegree = _this$props.gapDegree,
          gapPosition = _this$props.gapPosition;
      var radius = 50 - strokeWidth / 2;
      var beginPositionX = 0;
      var beginPositionY = -radius;
      var endPositionX = 0;
      var endPositionY = -2 * radius;

      switch (gapPosition) {
        case 'left':
          beginPositionX = -radius;
          beginPositionY = 0;
          endPositionX = 2 * radius;
          endPositionY = 0;
          break;

        case 'right':
          beginPositionX = radius;
          beginPositionY = 0;
          endPositionX = -2 * radius;
          endPositionY = 0;
          break;

        case 'bottom':
          beginPositionY = radius;
          endPositionY = 2 * radius;
          break;

        default:
      }

      var pathString = "M 50,50 m ".concat(beginPositionX, ",").concat(beginPositionY, "\n     a ").concat(radius, ",").concat(radius, " 0 1 1 ").concat(endPositionX, ",").concat(-endPositionY, "\n     a ").concat(radius, ",").concat(radius, " 0 1 1 ").concat(-endPositionX, ",").concat(endPositionY);
      var len = Math.PI * 2 * radius;
      var trailPathStyle = {
        strokeDasharray: "".concat(len - gapDegree, "px ").concat(len, "px"),
        strokeDashoffset: "-".concat(gapDegree / 2, "px"),
        transition: 'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s'
      };
      var strokePathStyle = {
        strokeDasharray: "".concat(percent / 100 * (len - gapDegree), "px ").concat(len, "px"),
        strokeDashoffset: "-".concat(gapDegree / 2, "px"),
        transition: 'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s' // eslint-disable-line

      };
      return {
        pathString: pathString,
        trailPathStyle: trailPathStyle,
        strokePathStyle: strokePathStyle
      };
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          prefixCls = _this$props2.prefixCls,
          strokeWidth = _this$props2.strokeWidth,
          trailWidth = _this$props2.trailWidth,
          strokeColor = _this$props2.strokeColor,
          percent = _this$props2.percent,
          trailColor = _this$props2.trailColor,
          strokeLinecap = _this$props2.strokeLinecap,
          style = _this$props2.style,
          className = _this$props2.className,
          restProps = _objectWithoutProperties(_this$props2, ["prefixCls", "strokeWidth", "trailWidth", "strokeColor", "percent", "trailColor", "strokeLinecap", "style", "className"]);

      var _this$getPathStyles = this.getPathStyles(),
          pathString = _this$getPathStyles.pathString,
          trailPathStyle = _this$getPathStyles.trailPathStyle,
          strokePathStyle = _this$getPathStyles.strokePathStyle;

      delete restProps.percent;
      delete restProps.gapDegree;
      delete restProps.gapPosition;
      return React.createElement("svg", {
        className: "".concat(prefixCls, "-circle ").concat(className),
        viewBox: "0 0 100 100",
        style: style
      }, React.createElement("path", {
        className: "".concat(prefixCls, "-circle-trail"),
        d: pathString,
        stroke: trailColor,
        strokeWidth: trailWidth || strokeWidth,
        fillOpacity: "0",
        style: trailPathStyle
      }), React.createElement("path", {
        className: "".concat(prefixCls, "-circle-path"),
        d: pathString,
        strokeLinecap: strokeLinecap,
        stroke: strokeColor,
        strokeWidth: this.props.percent === 0 ? 0 : strokeWidth,
        fillOpacity: "0",
        style: strokePathStyle
      }));
    }
  }]);

  return Circle;
}(React.Component);

_defineProperty(Circle, "defaultProps", {
  className: '',
  percent: 0,
  gapDegree: 0,
  gapPosition: 'top',
  prefixCls: 'panda-progress',
  strokeColor: '#2db7f5',
  strokeLinecap: 'round',
  strokeWidth: 1,
  style: {},
  trailColor: '#D9D9D9',
  trailWidth: 1
});

export default Circle;