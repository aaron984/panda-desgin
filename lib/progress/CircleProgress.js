"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var Circle =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2["default"])(Circle, _React$Component);

  function Circle() {
    (0, _classCallCheck2["default"])(this, Circle);
    return (0, _possibleConstructorReturn2["default"])(this, (0, _getPrototypeOf2["default"])(Circle).apply(this, arguments));
  }

  (0, _createClass2["default"])(Circle, [{
    key: "getPathStyles",
    value: function getPathStyles() {
      var _this$props = this.props,
          percent = _this$props.percent,
          strokeWidth = _this$props.strokeWidth,
          gapDegree = _this$props.gapDegree,
          gapPosition = _this$props.gapPosition;
      var radius = 50 - strokeWidth / 2;
      var beginPositionX = 0;
      var beginPositionY = -radius;
      var endPositionX = 0;
      var endPositionY = -2 * radius;

      switch (gapPosition) {
        case 'left':
          beginPositionX = -radius;
          beginPositionY = 0;
          endPositionX = 2 * radius;
          endPositionY = 0;
          break;

        case 'right':
          beginPositionX = radius;
          beginPositionY = 0;
          endPositionX = -2 * radius;
          endPositionY = 0;
          break;

        case 'bottom':
          beginPositionY = radius;
          endPositionY = 2 * radius;
          break;

        default:
      }

      var pathString = "M 50,50 m ".concat(beginPositionX, ",").concat(beginPositionY, "\n     a ").concat(radius, ",").concat(radius, " 0 1 1 ").concat(endPositionX, ",").concat(-endPositionY, "\n     a ").concat(radius, ",").concat(radius, " 0 1 1 ").concat(-endPositionX, ",").concat(endPositionY);
      var len = Math.PI * 2 * radius;
      var trailPathStyle = {
        strokeDasharray: "".concat(len - gapDegree, "px ").concat(len, "px"),
        strokeDashoffset: "-".concat(gapDegree / 2, "px"),
        transition: 'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s'
      };
      var strokePathStyle = {
        strokeDasharray: "".concat(percent / 100 * (len - gapDegree), "px ").concat(len, "px"),
        strokeDashoffset: "-".concat(gapDegree / 2, "px"),
        transition: 'stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s' // eslint-disable-line

      };
      return {
        pathString: pathString,
        trailPathStyle: trailPathStyle,
        strokePathStyle: strokePathStyle
      };
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          prefixCls = _this$props2.prefixCls,
          strokeWidth = _this$props2.strokeWidth,
          trailWidth = _this$props2.trailWidth,
          strokeColor = _this$props2.strokeColor,
          percent = _this$props2.percent,
          trailColor = _this$props2.trailColor,
          strokeLinecap = _this$props2.strokeLinecap,
          style = _this$props2.style,
          className = _this$props2.className,
          restProps = (0, _objectWithoutProperties2["default"])(_this$props2, ["prefixCls", "strokeWidth", "trailWidth", "strokeColor", "percent", "trailColor", "strokeLinecap", "style", "className"]);

      var _this$getPathStyles = this.getPathStyles(),
          pathString = _this$getPathStyles.pathString,
          trailPathStyle = _this$getPathStyles.trailPathStyle,
          strokePathStyle = _this$getPathStyles.strokePathStyle;

      delete restProps.percent;
      delete restProps.gapDegree;
      delete restProps.gapPosition;
      return _react["default"].createElement("svg", {
        className: "".concat(prefixCls, "-circle ").concat(className),
        viewBox: "0 0 100 100",
        style: style
      }, _react["default"].createElement("path", {
        className: "".concat(prefixCls, "-circle-trail"),
        d: pathString,
        stroke: trailColor,
        strokeWidth: trailWidth || strokeWidth,
        fillOpacity: "0",
        style: trailPathStyle
      }), _react["default"].createElement("path", {
        className: "".concat(prefixCls, "-circle-path"),
        d: pathString,
        strokeLinecap: strokeLinecap,
        stroke: strokeColor,
        strokeWidth: this.props.percent === 0 ? 0 : strokeWidth,
        fillOpacity: "0",
        style: strokePathStyle
      }));
    }
  }]);
  return Circle;
}(_react["default"].Component);

(0, _defineProperty2["default"])(Circle, "defaultProps", {
  className: '',
  percent: 0,
  gapDegree: 0,
  gapPosition: 'top',
  prefixCls: 'panda-progress',
  strokeColor: '#2db7f5',
  strokeLinecap: 'round',
  strokeWidth: 1,
  style: {},
  trailColor: '#D9D9D9',
  trailWidth: 1
});
var _default = Circle;
exports["default"] = _default;