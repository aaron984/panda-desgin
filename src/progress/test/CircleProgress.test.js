import React from 'react';
import { render, mount } from 'enzyme';
import CircleProgress from '../CircleProgress';

describe('CircleProgress', () => {
  it('CircleProgress gapPosition equal to left renders correctly', () => {
    const wrapper = render(<CircleProgress trailWidth={0} gapPosition="left" percent={12} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('CircleProgress gapPosition equal to right renders correctly', () => {
    const wrapper = render(<CircleProgress trailWidth={0} gapPosition="right" percent={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
