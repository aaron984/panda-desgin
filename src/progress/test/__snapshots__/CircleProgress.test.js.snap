// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleProgress CircleProgress gapPosition equal to left renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "panda-progress-circle ",
      "viewBox": "0 0 100 100",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "panda-progress-circle-trail",
          "d": "M 50,50 m -49.5,0
     a 49.5,49.5 0 1 1 99,0
     a 49.5,49.5 0 1 1 -99,0",
          "fill-opacity": "0",
          "stroke": "#D9D9D9",
          "stroke-width": "1",
          "style": "stroke-dasharray:311.01767270538954px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Object {
          "attribs": Object {
            "class": "panda-progress-circle-path",
            "d": "M 50,50 m -49.5,0
     a 49.5,49.5 0 1 1 99,0
     a 49.5,49.5 0 1 1 -99,0",
            "fill-opacity": "0",
            "stroke": "#2db7f5",
            "stroke-linecap": "round",
            "stroke-width": "1",
            "style": "stroke-dasharray:37.322120724646744px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-linecap": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-linecap": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "panda-progress-circle-path",
          "d": "M 50,50 m -49.5,0
     a 49.5,49.5 0 1 1 99,0
     a 49.5,49.5 0 1 1 -99,0",
          "fill-opacity": "0",
          "stroke": "#2db7f5",
          "stroke-linecap": "round",
          "stroke-width": "1",
          "style": "stroke-dasharray:37.322120724646744px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "panda-progress-circle-trail",
            "d": "M 50,50 m -49.5,0
     a 49.5,49.5 0 1 1 99,0
     a 49.5,49.5 0 1 1 -99,0",
            "fill-opacity": "0",
            "stroke": "#D9D9D9",
            "stroke-width": "1",
            "style": "stroke-dasharray:311.01767270538954px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-linecap": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-linecap": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "viewBox": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "viewBox": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`CircleProgress CircleProgress gapPosition equal to right renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "panda-progress-circle ",
      "viewBox": "0 0 100 100",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "panda-progress-circle-trail",
          "d": "M 50,50 m 49.5,0
     a 49.5,49.5 0 1 1 -99,0
     a 49.5,49.5 0 1 1 99,0",
          "fill-opacity": "0",
          "stroke": "#D9D9D9",
          "stroke-width": "1",
          "style": "stroke-dasharray:311.01767270538954px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Object {
          "attribs": Object {
            "class": "panda-progress-circle-path",
            "d": "M 50,50 m 49.5,0
     a 49.5,49.5 0 1 1 -99,0
     a 49.5,49.5 0 1 1 99,0",
            "fill-opacity": "0",
            "stroke": "#2db7f5",
            "stroke-linecap": "round",
            "stroke-width": "0",
            "style": "stroke-dasharray:0px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-linecap": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-linecap": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "panda-progress-circle-path",
          "d": "M 50,50 m 49.5,0
     a 49.5,49.5 0 1 1 -99,0
     a 49.5,49.5 0 1 1 99,0",
          "fill-opacity": "0",
          "stroke": "#2db7f5",
          "stroke-linecap": "round",
          "stroke-width": "0",
          "style": "stroke-dasharray:0px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
        },
        "children": Array [],
        "name": "path",
        "namespace": "http://www.w3.org/2000/svg",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "panda-progress-circle-trail",
            "d": "M 50,50 m 49.5,0
     a 49.5,49.5 0 1 1 -99,0
     a 49.5,49.5 0 1 1 99,0",
            "fill-opacity": "0",
            "stroke": "#D9D9D9",
            "stroke-width": "1",
            "style": "stroke-dasharray:311.01767270538954px 311.01767270538954px;stroke-dashoffset:-0px;transition:stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s",
          },
          "children": Array [],
          "name": "path",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "d": undefined,
            "fill-opacity": undefined,
            "stroke": undefined,
            "stroke-width": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-linecap": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "d": undefined,
          "fill-opacity": undefined,
          "stroke": undefined,
          "stroke-linecap": undefined,
          "stroke-width": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "svg",
    "namespace": "http://www.w3.org/2000/svg",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "viewBox": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "viewBox": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
